lib/child_process.js

```diff
@@ -91,7 +94,9 @@ exports.fork = function(modulePath /*, args, options*/) {
     options.stdio = options.silent ? stdioStringToArray('pipe') :
       stdioStringToArray('inherit');
   } else if (options.stdio.indexOf('ipc') === -1) {
-    throw new TypeError('Forked processes must have an IPC channel');
+    throw new errors.Error('ERR_CHILD_PROCESS_IPC_REQUIRED',
+                           'options.stdio',
+                           options.stdio);
```

Comment:
The error message looks like this: `Forked processes must have an IPC channel options.stdio stdioValue`. Is it intended? Maybe just using `ERR_INVALID_OPT_VALUE` is enough?

Reply:
Personally I prefer a separate error code -- I see the issue as not just "invalid option for stdio", but "forked process cannot be started because there's no ipc channel"

Reply:
I'm OK with a separate code, but the message still looks weird